clang/lib/Sema/SemaType.cpp

```diff
@@ -2527,7 +2527,7 @@ QualType Sema::BuildMatrixType(QualType ElementTy, Expr *NumRows, Expr *NumCols,
 }
 
 bool Sema::CheckFunctionReturnType(QualType T, SourceLocation Loc) {
-  if (T->isArrayType() || T->isFunctionType()) {
+  if ((!getLangOpts().HLSL && T->isArrayType()) || T->isFunctionType()) {
```

Comment:
Is this supposed to be here? This looks unrelated to splatting.

Reply:
Ooops!